src/integrationtests/java/com/aws/greengrass/integrationtests/deployment/DeploymentConfigMergingTest.java

```diff
@@ -217,6 +234,45 @@ void GIVEN_kernel_running_single_service_WHEN_merge_changes_service_THEN_service
         }
     }
 
+    @Test
+    void GIVEN_kernel_running_single_service_WHEN_merge_config_with_bootstrap_THEN_dependency_watcher_is_removed()
```

Comment:
Why were we not able to make a test for the kernel update workflow and have to try the regular deployment merge?

Reply:
Ok, looks like you are trying to merge something with bootstrap so it should take the bootstrap route, but what is this test doing? I don't see any verifications after the merge is performed

Reply:
I was using this test to investigate the root cause of this problem. Since KernelUpdateActivator eventually calls System.exit, I dont think I can use this integ test for any verification and planning to remove this. The plan is to instead test the E2E with a UAT. But I have kept the test here for now as a draft.